ml-agents/mlagents/trainers/model_saver/torch_model_saver.py

```diff
@@ -89,7 +89,25 @@ def _load_model(
         policy = cast(TorchPolicy, policy)
 
         for name, mod in modules.items():
-            mod.load_state_dict(saved_state_dict[name])
+            try:
+                mod.load_state_dict(saved_state_dict[name])
+            except Exception:
+                if name in saved_state_dict:
+                    logger.warning(f"Failed to load directly for module {name}.")
```

Comment:
```suggestion
                    logger.warning(f"Failed to load directly for module {name}, will attempt partial load.")
```

Or something similar

Reply:
This case also catches things that were not present in the previous run e.g. a newly added GAIL head. So it may not have been 'changed'